test/terra/backends/qasm_simulator/qasm_noise.py

```diff
@@ -148,3 +150,26 @@ def test_kraus_gate_noise(self):
                 **self.BACKEND_OPTS).result()
             self.assertSuccess(result)
             self.compare_counts(result, [circuit], [target], delta=0.05 * shots)
+
+class QasmKrausNoiseOnQFTTests:
+    """QasmSimulator Kraus error noise model on QFT tests."""
+
+    SIMULATOR = QasmSimulator()
+    BACKEND_OPTS = {}
+
+    def test_kraus_gate_noise_on_QFT(self):
+        """Test Kraus noise on a QFT circuit"""
+
+        shots = 1000
+        noise_models = ref_kraus_noise.kraus_gate_error_noise_models_full()
+
+        for noise_model in noise_models:
+            circuit = QFT(3)
+            ref_target = qi.Statevector(circuit).data
+            circuit.save_statevector()
+            job = execute([circuit], self.SIMULATOR, shots=shots,
+                          noise_model=noise_model,
+                          **self.BACKEND_OPTS)
+            result = job.result()
+            self.assertSuccess(result)
+            self.compare_statevector(result, [circuit], [ref_target])
```

Comment:
I don't understand.
1. How comes that the MPS simulator, which is a counts simulator, has a statevector as a result? (the function `compare_statevector` calls `result.get_statevector`, I expect this call to fail).
2. Is the statevector at the end of a noisy simulation expected to be the same as in a noiseless one?

Reply:
This comment was very helpful - I actually was running a model without noise in this test. I now corrected the model and used hard-coded reference values. Thanks!

Reply:
Great, it looks fine now